tests/test_ide_integration.py

```diff
@@ -0,0 +1,112 @@
+#    Copyright 2018 D-Wave Systems Inc.
+
+#    Licensed under the Apache License, Version 2.0 (the "License")
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+
+#        http: // www.apache.org/licenses/LICENSE-2.0
+
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+
+import unittest
+# import pytest
+import logging
+import os
+from subprocess import Popen, PIPE,STDOUT
+import inspect
+import time
+
+# These are integration tests with leap ide
+# This require loding leap ide docker image and run inside that image
+class LeapIdeIntegrationTests(unittest.TestCase):
+
+    def setUp(self):
+        self.verificationErrors = []
+        self.logger = None
+        self.log = self.custom_logger(logging.DEBUG)
+
+    def custom_logger(self,log_level=logging.DEBUG):
+
+        # Check if logger object exists
+        if self.logger is not None:
+            return self.logger
+
+        # Gets the name of the class / method from where this method is called
+        logger_name = inspect.stack()[1][3]
+        self.logger = logging.getLogger(logger_name)
+
+        # By default, log all messages
+        self.logger.setLevel(logging.DEBUG)
+
+        # Get log file name from environment getting set in testRunner.sh
+        log_file= os.getcwd()+"logs.log"
+        if os.getenv('log_file') is not None and os.getenv('log_file'):
+            log_file = os.environ['log_file']
+
+        # File handler for logging
+        file_handler = logging.FileHandler(log_file, mode='a')
+        file_handler.setLevel(log_level)
+        formatter = logging.Formatter('%(asctime)s - %(name)s - %(levelname)s: %(message)s', datefmt='%m/%d/%Y %I:%M:%S %p')
+        file_handler.setFormatter(formatter)
+        self.logger.addHandler(file_handler)
+
+        # Log to console
+        console_handler = logging.StreamHandler()
+        console_handler.setFormatter(formatter)
+        console_handler.setLevel(log_level)
+        self.logger.addHandler(console_handler)
+
+        return self.logger
+
+    def VerifyErrors(self,output):
+
+        try: 
+            self.assertNotIn("ERROR",output.upper() )
+            self.log.info("Verify if error string contains in output successful \n")
+        except AssertionError as e:
+            self.log.error("Verify if error string contains in output failed \n")
+            self.verificationErrors.append(str(e))
```

Comment:
You're using `unittest.TestCase`. You don't need to do try-except to test for errors, `unittest` will handle it for you. You could just replace it with a one-liner.

```
self.assertNotIn("ERROR", output.upper(), msg="Some error message")
```

Is it necessary to output the "successful"-case string message? (i.e. wouldn't a well-named-test-case passing be sufficient?)